sdk/identity/azure-identity/inc/azure/identity/azure_cli_credential.hpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Azure CLI Credential uses Azure CLI to obtain an access token.
+ */
+
+#pragma once
+
+#include <azure/core/credentials/credentials.hpp>
+#include <azure/core/credentials/token_credential_options.hpp>
+
+#include <azure/core/datetime.hpp>
+
+#include <chrono>
+#include <string>
+
+namespace Azure { namespace Identity {
+  /**
+   * @brief Options for configuring the #Azure::Identity::AzureCliCredential.
+   */
+  struct AzureCliCredentialOptions final : public Core::Credentials::TokenCredentialOptions
+  {
+  public:
+    /**
+     * @brief The ID of the tenant to which the credential will authenticate by default. If not
+     * specified, the credential will authenticate to any requested tenant, and will default to the
+     * tenant provided to the 'az login' command.
+     */
+    std::string TenantId;
+
+    /**
+     * @brief The CLI process timeout.
+     */
+    DateTime::duration CliProcessTimeout
+        = std::chrono::seconds(13); // Value was taken from .NET SDK.
+  };
+
+  /**
+   * @brief Enables authentication to Azure Active Directory using Azure CLI to obtain an access
+   * token.
+   */
+  class AzureCliCredential
+#if !defined(TESTING_BUILD)
+      final
+#endif
+      : public Core::Credentials::TokenCredential {
+  protected:
```

Comment:
Note that the "protected:" should also be covered via `if !defined(TESTING_BUILD)` - that's because ApiView has a diagnostic on protected fields in a class marked "final".